spotirec.py

```diff
@@ -236,25 +237,75 @@ def add_custom_seed_info(data: json):
         rec.add_seed_info(data_dict=data['items'][int(x)])
 
 
+def request_data(uri: str, data_type: str) -> json:
+    """
+    Requests data about an artist or a track.
+    :param uri: uri for the artist or track
+    :param data_type: the type of data to request; 'artists' or 'tracks'
+    :return: data about artist or track as a json obj
+    """
+    response = requests.get(f'{url_base}/{data_type}/{uri.split(":")[2]}', headers=headers)
+    return json.loads(response.content.decode('utf-8'))
+
+
+def add_to_blacklist(entries: list):
+    """
+    Add input uris to blacklist and exit
+    :param entries: list of input uris
+    """
+    with open(blacklist_path, 'r') as file:
+        try:
+            data = json.loads(file.read())
+        except json.decoder.JSONDecodeError:
+            data = {'tracks': [],
+                    'artists': []}
+        for uri in entries:
+            if 'track' in uri:
+                track = request_data(uri, 'tracks')
+                artists = [x['name'] for x in track['artists']]
+                data['tracks'].append({'name': track['name'],
+                                       'uri': uri,
+                                       'artists': artists})
+                print(f'Added track \"{track["name"]}\" by {", ".join(str(x) for x in artists).strip(", ")}'
+                      f' to your blacklist')
+            elif 'artist' in uri:
+                artist = request_data(uri, 'artists')
+                data['artists'].append({'name': artist['name'],
+                                        'uri': artist['uri']})
+                print(f'Added artist \"{artist["name"]}\" to your blacklist')
+            else:
+                print(f'uri \"{uri}\" is either not a valid uri for a track or artist, or is malformed and has '
+                      f'not been added to the blacklist')
+    with open(blacklist_path, 'w+') as file:
+        file.write(json.dumps(data))
```

Comment:
Are we still assuming that blacklist exists? I think that this should be defensively coded against, such that we try-except and exit early with a status message.

Reply:
No. This was fixed in an earlier commit unrelated to this PR.

Line 40-42 in `spotirec.py` on current master branch:
```python
if not os.path.exists(blacklist_path):
    f = open(blacklist_path, 'w')
    f.close()
```